src/parser/processor.ts

```diff
@@ -0,0 +1,81 @@
+import * as fs from "fs";
+import configuration from "../configuration/config-reader";
+import { GetActivity } from "../get-activity";
+import program from "./command-line";
+
+export class Processor {
+  private _transformers: Array<Transformer> = [];
+  private _result: Result = {};
+  private readonly _configuration = configuration;
+
+  add(transformer: Transformer) {
+    this._transformers.push(transformer);
+    return this;
+  }
+  async run(data: Readonly<GetActivity>) {
+    if (this._configuration.disabled) {
+      console.log("Module is disabled. Skipping...");
+      return;
+    }
+    for (const transformer of this._transformers) {
+      if (transformer.enabled) {
+        this._result = await transformer.transform(data, this._result);
+      }
+    }
+    // Aggregate total result
+    for (const item of Object.keys(this._result)) {
+      this._result[item].total = this._sumRewards(this._result[item]);
+    }
+    return this._result;
+  }
+  dump() {
+    const { file } = program.opts();
+    const result = JSON.stringify(this._result, undefined, 2);
+    if (!file) {
+      console.log(result);
+    } else {
+      fs.writeFileSync(file, result);
+    }
+  }
+  _sumRewards(obj: Record<string, object | number>) {
```

Comment:
Generally speaking, using `object` as a data type is considered bad practice. Instead you should define the object shape with something like `Record<string, unknown>` for something really generic and temporary, but the more specific, the better.